common/compiler/solidity_test.go

```diff
@@ -70,9 +90,51 @@ func TestSolidityCompiler(t *testing.T) {
 func TestSolidityCompileError(t *testing.T) {
 	skipWithoutSolc(t)
 
+	// Force syntax error by removing some characters.
 	contracts, err := CompileSolidityString("", testSource[4:])
 	if err == nil {
 		t.Errorf("error expected compiling source. got none. result %v", contracts)
 	}
 	t.Logf("error: %v", err)
 }
+
+func TestSolcCanCompile(t *testing.T) {
+	solcVersion := "0.8.11"
+	tests := []struct {
+		version  string
+		expected bool
+	}{
+		{"^0.8.11", true},
+		{"^0.4.24", false},
+		{"^0.5.6", false},
+		{"0.5.6", false},
```

Comment:
Missing test cases may be added such as `pragma solidity >=0.6.0 <0.9.0;`

Reply:
I removed the solidity version parsing logic from the previous request.
This latst change just compiles the target contract with the available solc and loads bytecode if it failed.